### PR TITLE
Make regression problem

### LINKED ISSUE #36: Add sparse group regression problem generator
`datasets.py` has the `make_sparse_group_classification` function which is a sparse group extension of `sklearn.datasets.make_classification`. We should also make a sparse group regression problem generator

### PR DESCRIPTION
**Pull request by richford**
Resolves #36.

Obviates #41

This PR removes the duplicate make_classification function and relies on the sklearn version.

### CODE REVIEW

File: afqinsight/tests/test_datasets.py [hunk 1]
```diff
@@ -26,153 +30,216 @@ def test_make_classification():
...
-        2000,
-        "Unexpected number of unique rows",
-    )
+        .shape[0]
+        == 2000
+    ), "Unexpected number of unique rows"  # nosec
+
+    assert (np.unique(X.view([('', X.dtype)]*X.shape[1])).view(X.dtype)
+            .reshape(-1, X.shape[1]).shape[0] == 2000), (
+                "Unexpected number of unique rows")
```
Comment (richford) [inline afqinsight/tests/test_datasets.py:75]: Duplication of above test
```suggestion
```

**Comment by richford** [comment]
@mnarayan, let me know when you think this is ready for review.

**Comment by richford** [comment]
Maybe we should rename `make_sparse_group_regression` to `make_group_regression` and `make_sparse_group_classification` to `make_group_classification`. It's easier to type and also more accurate since the sparsity is really up to the user and the real functionality that we're adding is the grouping.

File: afqinsight/tests/test_datasets.py (hunk 1 above)
Comment (mnarayan) [reply afqinsight/tests/test_datasets.py]: Do we want to get rid of this? It does effectively test for the C-contiguous issue which we now pass.

**Comment by mnarayan** [comment]
> Maybe we should rename `make_sparse_group_regression` to `make_group_regression` and `make_sparse_group_classification` to `make_group_classification`. It's easier to type and also more accurate since the sparsity is really up to the user and the real functionality that we're adding is the grouping.

Sounds good. I will make this change.

File: afqinsight/tests/test_datasets.py (hunk 1 above)
Comment (richford) [reply afqinsight/tests/test_datasets.py]: Agreed. It just looks like a duplicate of the assert statement directly above (albeit with some formatting differences).

**Comment by richford** [comment]
Thanks @mnarayan. I removed the "WIP" from the PR title and will merge after tests pass.

**Pull request merged**